omod/src/main/webapp/resources/scripts/components/LabTrackingOrderFactory.js

```diff
@@ -333,10 +347,11 @@ angular.module("labTrackingOrderFactory", [])
                 concept: LabTrackingOrder.concepts.order.conceptUuid,
                 careSetting: labTrackingOrder.careSetting.value,
                 encounter: labTrackingOrder.encounter.value,
-                orderReason: labTrackingOrder.preLabDiagnosis.value,
+                orderReason: typeof labTrackingOrder.preLabDiagnosis === 'object' ? labTrackingOrder.preLabDiagnosis.value : null,
+                orderReasonNonCoded: typeof labTrackingOrder.preLabDiagnosis === 'string' ? labTrackingOrder.preLabDiagnosis : null,
```

Comment:
The intent would be clearer if you branched based on the `type`, if that is available.